A bit better for country / city layers

kent1D opened the pull request:
We deal with an importance int to avoid overflowing at zoom 3

kent1D committed:
Add importance of type int (easier to filter)

add name:fr_short for short names

kent1D committed:
Cape Town is not a capital with embassy

kent1D committed:
Add importance as int for country layer

kent1D committed:
review a bit the rendering of #country layer

kent1D committed:
Better management of zooms

kent1D committed:
Review importance

kent1D committed:
Divide import in 4 functions

yohanboniface commented on labels.mss:
```diff
@@ -1,22 +1,29 @@
-#country[zoom>=3][zoom<=7] {
+#country[importance<=3][zoom>=3][zoom<=10],
```
I usually prevent overlap for the same source. Not sure how Mapnik deals with that.
So, I'd do:

    #country[importance<=3][zoom=3],
    #country[importance<=4][zoom=4],
    #country[zoom>=5][zoom<=10] {

kent1D replied:
ok thanks

kent1D replied:
Changed in new commit

yohanboniface commented on labels.mss:
```diff
@@ -1,22 +1,29 @@
   text-line-spacing: 0.1;
+  [zoom>=5] {
+    text-size: 14;
+    text-dx: 10;
```
I guess you removed default text-margin for "diplomatic" reasons? Because it ends with labels being very close, and thus the map being less readable. Just to be sure we *need* this "regression" ;)

![screenshot-20180213180123-574x410](https://user-images.githubusercontent.com/146023/36162872-eeef3636-10e7-11e8-9382-fc7b4cea8d91.png)

Can't we go with a more readable map and have this kind of adjustment if someone complain? :)

kent1D replied:
Mmmh just bad cut / paste for the text-dx and text-dy

Sorry, I agree completely, I even increased it a bit at small zoom levels

yohanboniface replied:
👍

yohanboniface commented on labels.mss:
```diff
@@ -46,8 +53,8 @@
-#city[zoom>=7][zoom<=10],
-#place_low[capital='yes'][zoom>=5][zoom<=10],
+#city[zoom>=5][zoom<=10],
+#place_low[capital='yes'][zoom>=7][zoom<=10],
```
I think this row could actually be removed: I don't think we want to deal with capitals which are not in our own data file anyway.

kent1D replied:
+1 but if we have capitals in place_low that are not in our file they won't appear as city, so we should add them in our file without the capital flag, isn't it?

yohanboniface replied:
Oh, actually, now that we start at zoom 7 instead of 5 this row is exactly redundant with the one below ;)

yohanboniface commented on piano.yml:
```diff
@@ -275,7 +275,7 @@ Layer:
     table: |-
         ( SELECT
-            geometry, COALESCE("name:{lang}", name) AS name, '{lang}' AS lang
+            geometry, COALESCE("name:{lang}", name) AS name, '{lang}' AS lang, CAST("importance" as INTEGER)
```
Shorter syntax for casting:

    SELECT importance::int

yohanboniface commented on data/country.csv:
```diff
@@ -1,218 +1,218 @@
-British Indian Ocean Territory,Territoire britannique de l'Océan Indien,British Indian Ocean Territory,إقليم المحيط الهندي البريطاني,99-Other,-6.3412369,71.8692598,IO
-Falkland Islands,Îles Malouines (Îles Falkland),Falkland Islands,جزر الفولكلاند,99-Other,-51.9579887,-59.569457,FK
-San Marino,Saint-Marin,San Marino,,99-Other,43.9428966,12.4596955,SM
+name,name:fr,name:fr_short,name:en,name:ar,importance,type,latitude,longitude,iso
```
Why keeping type AND importance?

kent1D replied:
type is what it is, importance is more or less similar now but permits to play a bit more. For example to avoid holes in big surfaces (name of countries not showed but there is lots of space around, which is not the case everywhere).

yohanboniface replied:
Not sure a "numeric" type is really useful but ok. :)

kent1D replied:
the use of "<=" in selectors didn't seem to work without it, don't know

yohanboniface replied:
>  the use of "<=" in selectors didn't seem to work without it, don't know

Humm, not sure to get this :s

I'm suggesting that a numeric value for type may not be a useful info given we also have a numeric value for "prio". I feel like we could deal only with "prio".

Are we using "type" somehow now?

kent1D replied:
Ah ok, i misunderstood.

At the moment we don't use type anymore. Just that I would like to keep it in case. Not to loose the data / info that may be interesting to use after.

But yes, we can now delete the numeric prefix value

yohanboniface replied:
Oooohhh, aaaahhhh!
Now I understand :)
Firefox is totally mixing up the CSV in the diff, see:

![screenshot-20180214075853-1356x220](https://user-images.githubusercontent.com/146023/36191355-fe9fdab6-115c-11e8-9821-114a8a7df7df.png)

The "," coming after the arabic column is rendered before with the following digit, but the "-xxx" part of the type if left alone. LTR and RTL in the same text really is a hard deal! :p

Anyway, you can leave it as it, no matter about the numeric "prefix"

kent1D committed:
Changes following comment
https://github.com/tilery/pianoforte/pull/22#discussion_r167931257

kent1D committed:
text-margin shouldn't have been removed

kent1D committed:
Use "prio" as field for priority like in place_low layer

yohanboniface commented on data/country.csv:
```diff
@@ -1,218 +1,218 @@
-British Indian Ocean Territory,Territoire britannique de l'Océan Indien,British Indian Ocean Territory,إقليم المحيط الهندي البريطاني,99-Other,-6.3412369,71.8692598,IO
-Falkland Islands,Îles Malouines (Îles Falkland),Falkland Islands,جزر الفولكلاند,99-Other,-51.9579887,-59.569457,FK
-San Marino,Saint-Marin,San Marino,,99-Other,43.9428966,12.4596955,SM
+name,name:fr,name:fr_short,name:en,name:ar,prio,type,latitude,longitude,iso
```
Missed that: if we want to keep the OSM naming convention, I suggest using `short_name:fr` instead.
See https://wiki.openstreetmap.org/wiki/Key:name
Are you planning to also add short names for other languages later on? :)

kent1D replied:
Yes I will, ok for the convention, I changed "importance" for "prio" this morning to do it like in #place_low

kent1D replied:
It's not used at the moment, just in the csv

kent1D committed:
name:fr_short > short_name:fr

kent1D committed:
Correction of really big mistakes done by automatic scripting

Adding a field "ldir" which will be usefull for label placement

kent1D committed:
load "ldir" field from database

kent1D committed:
If ldir is set, use it to define the direction

kent1D committed:
Add ldir to database

kent1D committed:
ldir is only on #city

kent1D committed:
Better data

yohanboniface commented on labels.mss:
```diff
@@ -78,8 +78,11 @@
   shield-unlock-image: true;
   shield-min-distance: 10;
-  [ldir!=''] {
+  #city[ldir!=''] {
```
I suggest two things:
- use a null value (by coercing in SQL: `NULLIF(ldir, '') as ldir`)
- also emit a `ldir` column for countries, so the final XML will be simpler (CartoCSS to XML can become quickly a mess when using partial subrules), no need for that to add it in the CSV, just create a dummy value in SQL `SELECT … NULL AS ldir`

Also, when you change the SQL, make sure you change it in both styles if needed (at lot of layers are duplicated for now) ;)

Note: I may work on a better factorization between the two styles Layers at some point, but given this is not a Cartocss feature, neither a Kosmtik feature and neither a YML feature, I preferred for now to see how we are going before investing time in developing a "include" feature in Kosmtik. :)

kent1D replied:
I think I made the necessary changes, tell me if not

Thanks for the suggestions and feedbacks

kent1D committed:
Add a prio field for city datas like for countries

Simplify type field for city datas to filter them

Order by prio for country and cities

kent1D committed:
#country and #place_low have NULL as ldir to keep .mss simpler
(following comment)

#city ldir is null by default

kent1D committed:
Report SQL modifications to forte

kent1D committed:
Sarrebruck in Germany

yohanboniface commented on labels.mss:
```diff
@@ -78,7 +78,7 @@
   shield-unlock-image: true;
   shield-min-distance: 10;
-  #city[ldir!=''] {
+  [ldir!=''] {
```
Mapnik may be smart enough, but I'd write `ldir!=null` here to clean :)

yohanboniface reviewed (approved):
👍

kent1D committed:
Adding some missing capitals

For info, adding a wikidata column in case we would like in the future
to retrieve other datas by scripting

kent1D committed:
!=null is better

yohanboniface commented:
Good for me! Good for you? :)

kent1D commented:
Yes, I will make an orther one today or tomorrow for the cities icons

kent1D commented:
But a separate one

yohanboniface commented:
Should I deploy and regenerate low zooms yet or I wait for that new PR?

kent1D commented:
I would say to wait for next one please
